commit b8b604640e111fba2b18a0f60dcda01e8cd3651b
Author: bertin <macportsuser@macports.org>
Date:   Thu May 22 23:41:24 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/patch-better-registry-error.diff

diff --git a/src/registry2.0/registry_util.tcl b/src/registry2.0/registry_util.tcl
index 65c6bb9f9..1327f4485 100644
--- a/src/registry2.0/registry_util.tcl
+++ b/src/registry2.0/registry_util.tcl
@@ -131,7 +131,7 @@ proc run_target {port target options} {
         return 0
     }
 
-    if {![catch {set mport [mportopen_installed [$port name] [$port version] [$port revision] [$port variants] $options]}]} {
+    if {![catch {set mport [mportopen_installed [$port name] [$port version] [$port revision] [$port variants] $options]} err]} {
         set failed 0
         if {[catch {mportexec $mport $target} result]} {
             ui_debug $::errorInfo
@@ -169,7 +169,7 @@ proc run_target {port target options} {
         }
     } else {
         ui_debug $::errorInfo
-        ui_warn "Failed to open Portfile from registry for $portspec"
+        ui_warn "Failed to open Portfile from registry for $portspec ($err); registry=[$port portfile]"
     }
     return 0
 }
